### PR TITLE
Added fields on OrderBook and Balance

### PR DESCRIPTION
**Pull request by cymp**
- Metadata on OrderBook
- Timestamp on Balance

### CODE REVIEW

File: xchange-core/src/main/java/org/knowm/xchange/dto/account/Balance.java [hunk 1]
```diff
@@ -34,6 +36,7 @@
   private final BigDecimal borrowed;
   private final BigDecimal withdrawing;
   private final BigDecimal depositing;
+  private Date timestamp;
```
Comment (walec51) [inline xchange-core/src/main/java/org/knowm/xchange/dto/account/Balance.java:39]: Balance should be a value object

final fields, no setters

File: xchange-core/src/main/java/org/knowm/xchange/dto/account/Balance.java [hunk 2]
```diff
@@ -275,26 +278,32 @@ public BigDecimal getDepositing() {
     return depositing;
   }
 
+  /**
+   * Returns the timestamp of the balance
```
Comment (walec51) [inline xchange-core/src/main/java/org/knowm/xchange/dto/account/Balance.java:282]: the is no sense in repeating the word timestamp in the docs

should the client set this field when the data was downloaded or should this information be from the exchanges server?

File: xchange-core/src/main/java/org/knowm/xchange/dto/marketdata/OrderBook.java [hunk 3]
```diff
@@ -131,6 +134,14 @@ public Date getTimeStamp() {
     return type == OrderType.ASK ? asks : bids;
   }
 
+  public Map<String, Serializable> getMetadata() {
+    return metadata;
+  }
+
+  public void setMetadata(Map<String, Serializable> metadata) {
+    this.metadata = metadata;
+  }
```
Comment (walec51) [inline xchange-core/src/main/java/org/knowm/xchange/dto/marketdata/OrderBook.java:143]: OrderBook should be a value object

File: xchange-core/src/main/java/org/knowm/xchange/dto/marketdata/OrderBook.java [hunk 4]
```diff
@@ -25,6 +26,8 @@
   private final List<LimitOrder> bids;
   /** the timestamp of the orderbook according to the exchange's server, null if not provided */
   private Date timeStamp;
+  /** the metadata, possibly provided by the api */
+  private Map<String, Serializable> metadata;
```
Comment (walec51) [inline xchange-core/src/main/java/org/knowm/xchange/dto/marketdata/OrderBook.java:30]: this data structure is to loose

if you need exchange specific fields then use methods from *Raw classes

File: xchange-core/src/main/java/org/knowm/xchange/dto/account/Balance.java (hunk 1 above)
Comment (cymp) [reply xchange-core/src/main/java/org/knowm/xchange/dto/account/Balance.java]: fixed

File: xchange-core/src/main/java/org/knowm/xchange/dto/account/Balance.java (hunk 2 above)
Comment (cymp) [reply xchange-core/src/main/java/org/knowm/xchange/dto/account/Balance.java]: fixed

File: xchange-core/src/main/java/org/knowm/xchange/dto/marketdata/OrderBook.java (hunk 4 above)
Comment (cymp) [reply xchange-core/src/main/java/org/knowm/xchange/dto/marketdata/OrderBook.java]: reverted

**Comment by cymp** [comment]
Thx @walec51, please review again.

File: xchange-core/src/main/java/org/knowm/xchange/dto/marketdata/OrderBook.java (hunk 3 above)
Comment (cymp) [reply xchange-core/src/main/java/org/knowm/xchange/dto/marketdata/OrderBook.java]: reverted

File: xchange-core/src/main/java/org/knowm/xchange/dto/marketdata/OrderBook.java (hunk 4 above)
Comment (cymp) [reply xchange-core/src/main/java/org/knowm/xchange/dto/marketdata/OrderBook.java]: reverted

**Comment by cymp** [comment]
forget about it, I'll make a builder

**Pull request closed without merging**